pkg/network/vmispec/hotplug_test.go

```diff
@@ -0,0 +1,129 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package vmispec_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+	"kubevirt.io/kubevirt/tests/libvmi"
+)
+
+const (
+	extraNetworkName           = "othernet"
+	extraNetworkAttachmentName = "othernad"
+)
+
+var _ = Describe("NetworksToHotplug", func() {
+	const (
+		nadName     = "nad1"
+		networkName = "n1"
+	)
+
+	DescribeTable("", func(vmi *v1.VirtualMachineInstance, networksToHotplug ...v1.Network) {
+		Expect(vmispec.NetworksToHotplug(vmi)).To(ConsistOf(networksToHotplug))
+	},
+		Entry("VMI without networks in spec does not have anything to hotplug", libvmi.NewAlpine()),
```

Comment:
The tests look good, the only thing I would try to improve is their description.
It feels too long and it would be nice to see a smaller direct description.

For example:
- with no networks in spec and status, there is nothing to hotplug.
- with a network in spec and status, there is nothing to hotplug.
- with a network in spec that is missing from the status, hotplug it.
- with two networks in spec, one missing from status, hotplug the missing one.

You will also need this scenario to be covered:
- with a network in status that is missing from spec, there is nothing to hotplug.

Reply:
Thanks, this is helpful!